### PR TITLE
[bug, testing] remove estimator state indeterminacy from SPECIAL INSTANCES

### PR DESCRIPTION
**Pull request by icfaust**
# Description
SPECIAL_INSTANCES only works with a single instance of sklearnex estimators. This dict is needed for specifically testing oneDAL supported sklearnex features.  By using a single instance, tests may impact one another. By using clone, a new estimator is created using the same parameters without state. Clone depends on the ```__init__``` parameters of the estimator only.

Changes proposed in this pull request:
- create a special dictionary object which will clone the special estimator instances for use
- use sklearn clone in ```__getitem__```

### CODE REVIEW

File: sklearnex/tests/_utils.py [hunk 1]
```diff
@@ -87,18 +88,26 @@ def _load_all_models(with_sklearnex=True, estimator=True):
...
-        LogisticRegression(solver="newton-cg"),
-    ]
-}
+class sklearn_clone_dict(dict):
+
+    def __getitem__(self, key):
+        return clone(super().__getitem__(key))
+
+
+SPECIAL_INSTANCES = sklearn_clone_dict(
```
Comment (samir-nasibli) [inline sklearnex/tests/_utils.py]:
```suggestion
class _sklearn_clone_dict(dict):

    def __getitem__(self, key):
        return clone(super().__getitem__(key))


SPECIAL_INSTANCES = _sklearn_clone_dict(
```

**Comment by icfaust** [comment]
private CI failures due to unrelated PCA issues (not in special_instances)

**Pull request merged**